subprojects/emcee/queue-api-models/src/main/kotlin/com/avito/emcee/queue/BucketResult.kt

```diff
@@ -0,0 +1,47 @@
+package com.avito.emcee.queue
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+public data class BucketResult(
+    @Json(name = "testDestination")
+    val device: DeviceConfiguration,
+    val unfilteredResults: List<UnfilteredResult>
+) {
+    @JsonClass(generateAdapter = true)
+    public data class UnfilteredResult(
+        val testEntry: TestEntry,
+        val testRunResults: List<TestRunResult>
+    ) {
+        @JsonClass(generateAdapter = true)
+        public data class TestRunResult(
+            val udid: String,
+            val duration: Int, // in seconds
+            val exceptions: List<Exception>,
+            val hostName: String,
+            val logs: List<Log>,
+            val startTime: StartTime,
+            val succeeded: Boolean
+        ) {
+            @JsonClass(generateAdapter = true)
+            public data class Exception(
+                val filePathInProject: String,
+                val lineNumber: Int,
+                val reason: String,
+                val relatedTestName: RelatedTestName,
+            )
+
+            @JsonClass(generateAdapter = true)
+            public data class Log(val contents: String)
+
+            @JsonClass(generateAdapter = true)
+            public data class RelatedTestName(val className: String, val methodName: String)
```

Comment:
```suggestion
            public data class TestName(val className: String, val methodName: String)
```
?

Don't see the knowledge about "relation" in the model itself.